AtlasSearch/AtlasSearch/Views/MainPage.xaml.cs

```diff
@@ -0,0 +1,18 @@
+﻿using AtlasSearch.ViewModels;
+
+namespace AtlasSearch.Views;
+
+public partial class MainPage : ContentPage
+{
+    public MainPage()
+    {
+        InitializeComponent();
+    }
+
+    protected override void OnAppearing()
+    {
+        base.OnAppearing();
+
+        _ = ((MainViewModel)BindingContext).InitializeSearch();
```

Comment:
You can do this with Behaviours from the CommunityToolkit if you prefer, like here: https://github.com/realm/realm-dotnet-samples/blob/24d165a5f3803ba4d62250d001b122323eb4d9a4/QuickJournalSync/Views/EntriesPage.xaml#L22C28-L22C28